internal/gen/messages_marshaler.go

```diff
@@ -200,6 +200,11 @@ nextField:
 				case messageIsWKT(value.Message):
 					// If the map value is a WKT, write the WKT.
 					g.writeWKTValue(field, value.Message, "v")
+
+				// Has the same behaviour as the g.messageHasMarshaler case but is a catch all for when the message has a fieldmask.
```

Comment:
I think actually we can merge the cases, i.e.
`case g.messageHasMarshaler(value.Message), g.messageHasFieldMask(value.Message):`.

Reply:
This won't work as the order matters here. When the field is of type `google.protobuf.FieldMask` it should go throught the `messageIsWKT` case. This is because this field should has a JSON (un)marshaler provided that we should call. However if we check for `hasFieldMask` it will also be true. 

Now we could check for `hasMarshaler` and `hasFieldMask` after checking for `messageIsWKT` but than we run into the issue that it can certain types will be seen as a wrapper while they have "custom" marshalers available which causes the generator to error out. Same goes with the other orders.

I'm not happy with how it looks, but it is seems to be the only solution that works without diving deep into the logic and changing other checks and generators.

Reply:
Moved it into a seperate function so that at least the maintainability is the same.